pkg/apis/policy/v1beta1/types.go

```diff
@@ -0,0 +1,159 @@
+	// OR if the traffic matches at least one NetworkPolicyIngressRule across all of the NetworkPolicy
+	// objects whose podSelector matches the pod.
+	// If this field is nil, this NetworkPolicy does not affect ingress to the selected pods.
+	// If this field is non-nil but contains no rules, this NetworkPolicy allows no traffic.
+	// If this field is non-nil and contains at least one rule, this NetworkPolicy allows any traffic
+	// which matches at least one of the NetworkPolicyIngressRules in this list.
+	Ingress []NetworkPolicyIngressRule `json:"ingress,omitempty" protobuf:"bytes,2,rep,name=ingress"`
+}
+
+// This NetworkPolicyIngressRule matches traffic if and only if the traffic matches both Ports AND From.
+type NetworkPolicyIngressRule struct {
+	// List of ports which should be made accessible on the pods selected by PodSelector.
+	// Each item in this list is combined using a logical OR.
+	// If this field is nil, this NetworkPolicyIngressRule matches all ports (traffic not restricted by Port).
+	// If this field is non-nil but contains no items, this NetworkPolicyIngressRule matches no ports (no traffic matches).
+	// If this field is non-nil and contains at least one item, then this NetworkPolicyIngressRule allows traffic
+	// only if the traffic matches at least one NetworkPolicyPort in the Ports list.
+	Ports []NetworkPolicyPort `json:"ports,omitempty" protobuf:"bytes,1,rep,name=ports"`
+
+	// List of sources which should be able to access the pods selected by PodSelector.
+	// Items in this list are combined using a logical OR operation.
+	// If this field nil, this NetworkPolicyIngressRule matches all sources (traffic not restricted by source).
+	// If this field is non-nil but contains no items, this NetworkPolicyIngressRule matches no sources (no traffic matches).
+	// If this field is non-nil and contains at least on item, this NetworkPolicyIngressRule allows traffic only if the
+	// traffic matches at least one NetworkPolicyPeer in the From list.
+	From []NetworkPolicyPeer `json:"from,omitempty" protobuf:"bytes,2,rep,name=from"`
+}
+
+type NetworkPolicyPort struct {
+	// The protocol (TCP or UDP) which traffic must match.
+	// If not defined, this field defaults to TCP.
+	Protocol *v1.Protocol `json:"protocol,omitempty" protobuf:"bytes,1,opt,name=protocol,casttype=k8s.io/kubernetes/pkg/api/v1.Protocol"`
+
+	// If specified, the port on the given protocol.  This can
+	// either be a numerical or named port.  If this field is nil,
+	// this NetworkPolicyPort matches all port names and numbers.
+	// If non-nil, only traffic on the specified protocol AND port
+	// will be matched by this NetworkPolicyPort.
+	Port *intstr.IntOrString `json:"port,omitempty" protobuf:"bytes,2,opt,name=port"`
+}
+
+type NetworkPolicyPeer struct {
+	// If 'Namespaces' is defined, 'Pods' must not be.
+	// This is a label selector which selects Pods in this namespace.
+	// This NetworkPolicyPeer matches any pods selected by this selector.
```

Comment:
s/NetworkPolicyPeer//

in general, avoid referencing type names because JSON doesn't have them, and avoid naming fields except in JSON notation